src/services/variables.ts

```diff
@@ -19,15 +20,22 @@ export const VAR_LOGS_FORMAT = 'logsFormat';
 export const VAR_LOGS_FORMAT_EXPR = '${logsFormat}';
 export const VAR_LINE_FILTER = 'lineFilter';
 export const VAR_LINE_FILTER_EXPR = '${lineFilter}';
-export const LOG_STREAM_SELECTOR_EXPR = `${VAR_LABELS_EXPR} ${VAR_PATTERNS_EXPR} ${VAR_LOGS_FORMAT_EXPR} ${VAR_LEVELS_EXPR} ${VAR_FIELDS_EXPR} ${VAR_LINE_FILTER_EXPR}`;
-export const LOG_VOLUME_STREAM_SELECTOR_EXPR = `${VAR_LABELS_EXPR} ${VAR_PATTERNS_EXPR} ${VAR_LOGS_FORMAT_EXPR} ${VAR_FIELDS_EXPR} ${VAR_LINE_FILTER_EXPR}`;
-
-export const PATTERNS_SAMPLE_SELECTOR_EXPR = `${VAR_LABELS_EXPR} ${VAR_PATTERNS_EXPR} ${VAR_LOGS_FORMAT_EXPR}`;
+export const LOG_STREAM_SELECTOR_EXPR = `{${VAR_LABELS_EXPR}} ${VAR_PATTERNS_EXPR} ${VAR_LOGS_FORMAT_EXPR} ${VAR_LEVELS_EXPR} ${VAR_FIELDS_EXPR} ${VAR_LINE_FILTER_EXPR}`;
+export const PATTERNS_SAMPLE_SELECTOR_EXPR = `{${VAR_LABELS_EXPR}} ${VAR_PATTERNS_EXPR} ${VAR_LOGS_FORMAT_EXPR}`;
 export const EXPLORATION_DS = { uid: VAR_DATASOURCE_EXPR };
 export const ALL_VARIABLE_VALUE = '$__all';
 export const LEVEL_VARIABLE_VALUE = 'detected_level';
 export const PATTERNS_TEXT_FILTER = 'patternsFilter';
 
+export function getLogsStreamSelector(options: {
+  labelExpressionToAdd?: string;
+  structuredMetadataToAdd?: string;
+  fieldExpressionToAdd?: string;
+}) {
+  const { labelExpressionToAdd = '', structuredMetadataToAdd = '', fieldExpressionToAdd = '' } = options;
+  return `{${VAR_LABELS_EXPR}${labelExpressionToAdd}} ${structuredMetadataToAdd} ${VAR_PATTERNS_EXPR} ${VAR_LOGS_FORMAT_EXPR} ${fieldExpressionToAdd} ${VAR_LEVELS_EXPR} ${VAR_FIELDS_EXPR} ${VAR_LINE_FILTER_EXPR}`;
+}
+
```

Comment:
Nit.
```suggestion
export function getLogsStreamSelector(labelExpressionToAdd = '', structuredMetadataToAdd = '', fieldExpressionToAdd = '') {
  return `{${VAR_LABELS_EXPR}${labelExpressionToAdd}} ${structuredMetadataToAdd} ${VAR_PATTERNS_EXPR} ${VAR_LOGS_FORMAT_EXPR} ${fieldExpressionToAdd} ${VAR_LEVELS_EXPR} ${VAR_FIELDS_EXPR} ${VAR_LINE_FILTER_EXPR}`;
}

```

Reply:
I dunno about this, there's no order to the props, I might have a fieldExpressionToAdd but no labelExpressionToAdd, I don't want to have to call the function with a bunch of empty strings in the function, i.e. getLogsStreamSelector('','',fieldExpression)

Reply:
I see your point, but very quickly you can use the same argument to make every function receive and object and not have multiple arguments.

Reply:
IMO if you have more then 2 optional params it should be an object

Reply:
I couldn't articulate it yesterday, but I was actually thinking about this:

```suggestion
export function getLogsStreamSelector({ labelExpressionToAdd = '', structuredMetadataToAdd = '', fieldExpressionToAdd = '' }) {
  return `{${VAR_LABELS_EXPR}${labelExpressionToAdd}} ${structuredMetadataToAdd} ${VAR_PATTERNS_EXPR} ${VAR_LOGS_FORMAT_EXPR} ${fieldExpressionToAdd} ${VAR_LEVELS_EXPR} ${VAR_FIELDS_EXPR} ${VAR_LINE_FILTER_EXPR}`;
}
```

Single deconstructed argument. `options` as an object was the source of discomfort.

Irrelevant, but a fun conversation to have IMO.

Reply:
🤦 yeah this is what we want, I'll update it next time I'm in here